src/custom/state/claim/hooks/index.ts

```diff
@@ -365,3 +366,83 @@ function _hasNoInputOrInputIsGreaterThanClaimAmount(
 ): input is Required<ClaimInput> {
   return !input.amount || JSBI.greaterThan(JSBI.BigInt(input.amount), JSBI.BigInt(claim.amount))
 }
+
+type LastAddress = string
+type ClaimAddressMapping = { [firstAddress: string]: LastAddress }
```

Comment:
```suggestion
type ClaimAddressMapping = Record<string, LastAddress>
```

Reply:
I agree, but it's all copied code. Wanted to not change what's not needed.

If you feel strongly I can completely refactor it.